### PR TITLE
feat(csharp): remove Newtonsoft JSON.NET

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

Remove the Newtonsoft JSON.NET deps and replace it by the new `System.Text.Json` from dotnet.

🎟 JIRA Ticket: [DI-1824](https://algolia.atlassian.net/browse/DI-1824)

### Changes included:
- Remove deps
- Add Custom Serializer
- Remove all `DataContract` and `DataMember` attrs.

[DI-1824]: https://algolia.atlassian.net/browse/DI-1824?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by millotp — COMMENTED** [review]
very nice initiative !

File: playground/csharp/Playground/Program.cs [hunk 1]
```diff
@@ -45,13 +45,13 @@
...
     await SearchPlayground.Run(config);
-    await ABTesting.Run(config);
-    await Analytics.Run(config);
-    await Insights.Run(config);
-    await Monitoring.Run(config);
-    await Personalization.Run(config);
-    await QuerySuggestions.Run(config);
-    await Recommend.Run(config);
-    await Ingestion.Run(config);
+    // await ABTesting.Run(config);
```
[the remark this answers is not in the document]
Comment (morganleroi) [reply playground/csharp/Playground/Program.cs]: You're too fast. Done at the same time.

File: templates/csharp/modelGeneric.mustache [hunk 2]
```diff
@@ -277,107 +276,4 @@
             sb.Append("}\n");
             return sb.ToString();
         }
-
-        /// <summary>
-        /// Returns the JSON string presentation of the object
-        /// </summary>
-        /// <returns>JSON string presentation of the object</returns>
-        public {{#parent}}{{^isArray}}{{^isMap}}override {{/isMap}}{{/isArray}}{{/parent}}{{^parent}}virtual {{/parent}}string ToJson()
```
[the remark this answers is not in the document]
Comment (morganleroi) [reply templates/csharp/modelGeneric.mustache:285]: Indeed. I'll re-add the `toJson` I just removed.

[the remark this answers is not in the document]
Comment (millotp) [reply templates/csharp/modelGeneric.mustache:285]: equals and hashcode no ?

[the remark this answers is not in the document]
Comment (morganleroi) [reply templates/csharp/modelGeneric.mustache:285]: they were not activated and generated.

[the remark this answers is not in the document]
Comment (morganleroi) [reply templates/csharp/modelGeneric.mustache:285]: adding equals and getHashcode.

**Review by millotp — APPROVED** [review]
huge !

**Review by millotp — APPROVED** [review]

**Pull request merged**